server/models/base_model.py

```diff
@@ -13,6 +13,61 @@ class BaseModelMixin(object):
     corresponding table
     """
 
+    def to_dict(self, includes=[], excludes=[]):
+        """
+        WARNING
+        This is to be called from an existing instance so it has
+        access to existing values
+
+
+        Take in a list of includes and/or excludes then return a
+        dictionary of fields and values. If includes is passed but
+        not excludes, it will return all fields that were in includes.
+        If excludes is passed but not includes, it will return all fields
+        except those in excludes.
+        If neither are passed, it will return ALL fields and values in the model
+        """
+        serialized_fields = {}
+        # Get all fields if neither excludes or includes is passed
+        if len(includes) == 0 and len(excludes) == 0:
+            for field in dir(self):
+                # Ignore dunder methods or private attributes
+                if "_" in  field:
+                    continue
+                # Make sure the class has the attribute
+                try:
+                    serialized_fields[field] = getattr(self, field)
+                except AttributeError:
+                    raise AttributeError("Tried to get field %s but %s doesn't exist") % (field, field)
+                except:
+                    raise Exception("Uncaught exception in %s to_dict()" % (self))
+            return serialized_fields
+
+        # Get all fields except what was excluded
+        elif len(excludes) > 0:
```

Comment:
I don't think you need to handle this case, if you remove the `len(excludes) == 0` from the above case, you can check if each field is in the excludes array. In the default case where the array is empty, it will return all fields

Reply:
Ahh, good point.